pkg/controller/cloud/node_controller.go

```diff
@@ -148,6 +148,14 @@ func (cnc *CloudNodeController) updateNodeAddress(node *v1.Node, instances cloud
 		return
 	}
 
+	//Node not present according the cloud provider, no need to update node address.
+	if _, err := instances.ExternalID(types.NodeName(node.Name)); err != nil {
```

Comment:
You should use `ensureNodeExistsByProviderIDOrExternalID(instances, node)` to check for if the node exists.

Reply:
Thanks your review, I agree it!